index.html

```diff
@@ -86,25 +86,311 @@ <h3>Use-case #4: Avoiding "Hall of Mirrors"</h3>
+        <section>
+          <h4><dfn>CaptureActionEventInit</dfn></h4>
+          <pre class="idl">
+            dictionary CaptureActionEventInit : EventInit {
+              CaptureAction action;
+            };
+          </pre>
+          <dl
+            data-link-for="CaptureActionEventInit"
+            data-dfn-for="CaptureActionEventInit"
+          >
+            <dt>
+              <dfn>action</dfn>
+            </dt>
+            <dd>
+              The {{CaptureAction}} to initialize the event with.
+            </dd>
+          </dl>
+        </section>
+      </section>
+    </section>
+    <section id="capturing-side-actions">
+      <h2>Capturing Side for Actions</h2>
+      <p>
+        Capturing applications can enumerate available [=capture actions=] that
+        are supported on the video track they have obtained, by using
+        {{MediaStreamTrack/getSupportedCaptureActions}}, and can trigger those
+        actions by using {{MediaStreamTrack/sendCaptureAction}}.</li>
+      </ol>
+      <section id="capture-handle-get-supported-actions">
+        <h3>Enumerating supported actions and triggering them</h3>
+        <p>
+          When a {{MediaStreamTrack}} is a video track derived from screen-capture
+          of a <a data-cite="SCREEN-CAPTURE#dfn-browser">browser</a>
+          <a data-cite="SCREEN-CAPTURE#dfn-display-surface">display surface</a>,
+          {{MediaStreamTrack/getSupportedCaptureActions}} returns the set of
+          available [=capture actions=], if any, supported by the captured
+          application associated with this video track.
+        </p>
+        <pre class="idl">
+          partial interface MediaStreamTrack {
+            sequence&lt;DOMString&gt; getSupportedCaptureActions();
+            Promise&lt;undefined&gt; sendCaptureAction(CaptureAction action);
+          };
+        </pre>
+        <dl data-link-for="MediaStreamTrack" data-dfn-for="MediaStreamTrack">
+          <dt>
+            <dfn>getSupportedCaptureActions</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST return [=this=]'
+            {{MediaDevices/[[AvailableCaptureActions]]}} if defined, or `[]` if not defined.</p>
+          </dd>
+          <dt>
+            <dfn>sendCaptureAction</dfn>
+          </dt>
+          <dd>
+            <p>When this method is invoked, the user agent MUST run the following steps:</p>
+            <ol>
+              <li>
+                If the [=relevant global object=] of [=this=] does not have
+                [=transient activation=], return a promise [=rejected=] with
+                {{InvalidStateError}}.
+              </li>
+              <li>
+                [=Consume user activation=].
+              </li>
+              <li>
+                Let |action| be the method's first argument.
+              </li>
+              <li>
+                If |action| is not in [=this=]' {{MediaDevices/[[AvailableCaptureActions]]}},
```

Comment:
Perhaps you'd consider an MVP that does not bother validating the action, and trusts that the capturee can **ignore** messages it has not declared as supported? Ignoring unsupported messages requires 0 lines of code.

We **should**, however, ensure that only valid `CaptureActions` are sent. This was a corollary of your approach (validation happened when declaring supported actions), but if you take my suggestion from the previous paragraph, then we should add this new, narrower validation of `actions is in CaptureActions`.

Reply:
> Perhaps you'd consider an MVP that does not bother validating the action, and trusts that the capturee can ignore messages it has not declared as supported? 

I think an application sending an unsupported action is a programming error web developers would want us to surface early, especially when it is trivial to do so.

Reply:
It would not be a programming error if it's in response to an asynchronous change in the set of actions the captured application declares itself as supporting. And please note that even if we disallow changing the set, navigation can cause to it being changed. I think the simplest thing is for **both sides** to ignore this - the capturer ignores the asynchronous issue, and the capturee ignoring the unsupported message.